src/clarity-angular/emphasis/alert/alert.ts

```diff
@@ -10,20 +10,41 @@ import {
     Output
 } from "@angular/core";
 
+// providers
+import { AlertIconAndTypesService } from "./providers/icon-and-types-service";
+
 @Component({
     selector: "clr-alert",
+    providers: [ AlertIconAndTypesService ],
     templateUrl: "./alert.html"
 })
 export class Alert {
+    constructor(public iconService: AlertIconAndTypesService) {
+    }
+
     @Input("clrAlertSizeSmall") isSmall: boolean = false;
-    @Input("clrAlertType") type: string = "alert-info";
     @Input("clrAlertClosable") closable: boolean = true;
     @Input("clrAlertAppLevel") isAppLevel: boolean = false;
 
     @Input("clrAlertClosed") _closed: boolean = false;
     @Output("clrAlertClosedChange") _closedChanged: EventEmitter<boolean> = new EventEmitter<boolean>(false);
 
-    private alertTypes: string[] = [ "alert-info", "alert-warning", "alert-danger", "alert-success"];
+    @Input("clrAlertType")
+    set alertType(val: string) {
+        this.iconService.alertType = val;
+    }
+    get alertType(): string {
+        return this.iconService.alertType;
+    }
+
+    @Input("clrAlertIcon")
+    set alertIconShape(value: string) {
+        this.iconService.alertIconShape = value;
+    }
```

Comment:
Ok, so I want to prefix what I'm about to write by saying I'm ok with this PR. Barring a few changes I posted in this review, I think it's good to go.

But I don't think forcing Clarity icons here is the best way to go. We've always tried to stay open and allow people to project anything most of the time, so that would be more consistent with other components. I'm a bit scared of the precedent this sets.